src/docfx/config/Config.cs

```diff
@@ -146,97 +146,111 @@ public static Config Load(string docsetPath, CommandLineOptions options, bool ex
+            }
             return exists;
         }
 
-        private static Config LoadCore(string docsetPath, string configPath, CommandLineOptions options, bool extend, RestoreMap restoreMap)
+        private static (List<Error>, Config) LoadCore(string docsetPath, string configPath, CommandLineOptions options, bool extend, RestoreMap restoreMap)
         {
             // Options should be converted to config and overwrite the config parsed from docfx.yml
+            var errors = new List<Error>();
             Config config = null;
-            try
-            {
-                var optionConfigObject = options?.ToJObject();
-                var configObject = ExpandAndNormalize(JsonUtility.Merge(LoadConfigObject(docsetPath, configPath, extend, restoreMap), optionConfigObject));
-                config = configObject.ToObject<Config>(JsonUtility.DefaultDeserializer);
-            }
-            catch (Exception e)
-            {
-                throw Errors.InvalidConfig(configPath, e.Message).ToException(e);
-            }
+            var optionConfigObject = options?.ToJObject();
 
-            Validate(config, docsetPath);
+            var (loadErrors, value) = LoadConfigObject(docsetPath, configPath, extend, restoreMap);
+            errors.AddRange(loadErrors);
 
-            return config;
+            var configObject = ExpandAndNormalize(JsonUtility.Merge(value, optionConfigObject));
+            var deserializeErrors = new List<Error>();
+
+            (deserializeErrors, config) = JsonUtility.ToObject<Config>(configObject);
+            errors.AddRange(deserializeErrors);
+
+            errors.AddRange(Validate(config, docsetPath));
+            return (errors, config);
         }
 
-        private static void Validate(Config config, string docsetPath)
+        private static List<Error> Validate(Config config, string docsetPath)
         {
-            ValidateLocale(config);
-            ValidateContributorConfig(config.Contribution, docsetPath);
+            var errors = new List<Error>();
+            errors.AddRange(ValidateLocale(config));
+            errors.AddRange(ValidateContributorConfig(config.Contribution, docsetPath));
+            return errors;
         }
 
-        private static void ValidateLocale(Config config)
+        private static List<Error> ValidateLocale(Config config)
         {
+            var errors = new List<Error>();
             try
             {
                 var culture = new CultureInfo(config.Locale);
             }
             catch (CultureNotFoundException e)
             {
-                throw Errors.InvalidLocale(config.Locale).ToException(e);
+                errors.Add(Errors.InvalidLocale(config.Locale));
             }
+            return errors;
         }
 
-        private static void ValidateContributorConfig(ContributionConfig config, string docsetPath)
+        private static List<Error> ValidateContributorConfig(ContributionConfig config, string docsetPath)
```

Comment:
can this be implemented as `ExternalFileAttribute` ?

Reply:
Will create a new PR for this

---
In reply to: [211462828](https://github.com/dotnet/docfx/pull/3243#discussion_r211462828) [](ancestors = 211462828)